src/result-decorator.test.ts

```diff
@@ -23,60 +17,97 @@ describe('Result Decorator', () => {
             timestamp: '100',
         } as AxeResults;
 
-        const wcagInfo: DictionaryStringTo<WCAG[]> = {};
-        const resultDecorator = new ResultDecorator(wcagInfo);
+        const irrelevantWcagInfo: DictionaryStringTo<WCAGLinkData> = {};
+        const resultDecorator = new ResultDecorator(irrelevantWcagInfo);
+
         const decoratedResults: DecoratedAxeResults = resultDecorator.decorateResults(
-            resultStub,
+            inputWithNoIndividualResults,
         );
-        expect(decoratedResults.violations).toEqual([]);
-        expect(decoratedResults).toMatchSnapshot();
+
+        expect(decoratedResults).toMatchObject({
+            timestamp: '100',
+            targetPageUrl: 'https://example.com',
+            targetPageTitle: '',
+        });
     });
 
-    it('result decorator contains WCAG information that is provided as dependency', () => {
+    // tslint:disable-next-line: mocha-no-side-effect-code
+    const resultsArrayProperties = [
+        'passes',
+        'violations',
+        'inapplicable',
+        'incomplete',
+    ] as const;
```

Comment:
What does as `const` do? I am a little confused by this.

Reply:
Const assertions tell TypeScript you're intending a deeper level of immutability than just the reference. [More details on TypeScript site](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions).

Reply:
This is a typescript 3.4 feature called a "const assertion". You can read full details in the [typescript 3.4 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html).

In this particular case, using it turns the type of resultsArrayProperties from `string[]` to the constant tuple type `readonly ["passes", "violations", "inapplicable", "incomplete"]`. This is useful here because it lets us avoid casting later; specifically, it enables lines 53 and 75 to understand that the type of `resultStub[resultsArrayPropertyUnderTest]` is going to specifically match the type of one of `resultsStub.passes`, `resultsStub.violations`, etc, rather than the weaker understanding that it will match the type of *some arbitrary* property of `resultsStub` (which is all the type checker could understand if it thought `resultsArrayPropertyUnderTest` was an arbitrary `string` as opposed to one of the 4 known constant elements of our constant tuple type).

Reply:
Thanks guys. Appreciate the links 👍